RFCs/0000-add-puppet-dev-flag.md

```diff
@@ -0,0 +1,63 @@
+# Add `--puppet-dev` flag to validate and test unit
+
+## Summary
+
+Introduces a new flag to the `validate` and `test unit` commands that provides a user, with public internet access, to validate and test against the latest puppet source code on GitHub.
+
+## Background & Assumptions
+
+Since the release of v1.5.0, PDK has been able to validate and test against various versions of the Puppet gem via the `--puppet-version` and `--pe-version` flags, however users anticipating upcoming Puppet releases may want to test and validate their modules against the unreleased [Puppet source](https://github.com/puppetlabs/puppet). This feature is geared towards giving module developers or consumers a head start on preparing their modules or installations for the new version of Puppet before it releases.
+
+## Motivation & Goals
+
+- Allow PDK Users to update their modules in preparation for upcoming releases of Puppet.
+- Minimizes the risk of upgrading established Puppet installations to new versions of Puppet.
+- Promotes more updated modules on the Forge immediately upon each Puppet version release.
+
+## Proposed Implementation Design
+
+Implement a new flag, `--puppet-dev`, to the `pdk test unit` and `pdk validate` subcommands. This flag will only be available to users with access to public internet and is able to access [GitHub](https://github.com). PDK installations in air-gapped or firewalled networks attempting to use this flag will result in a connectivity error, resulting in a non-zero exit code.
```

Comment:
I think we should add support for `PDK_PUPPET_DEV` to control this as well, since we already have `PDK_PUPPET_VERSION` and `PDK_PE_VERSION`.